src/sagemaker/estimator.py

```diff
@@ -599,10 +601,33 @@ def __init__(
         self.output_kms_key = resolve_value_from_config(
             output_kms_key, TRAINING_JOB_KMS_KEY_ID_PATH, sagemaker_session=self.sagemaker_session
         )
-        self.volume_kms_key = resolve_value_from_config(
-            volume_kms_key,
-            TRAINING_JOB_VOLUME_KMS_KEY_ID_PATH,
-            sagemaker_session=self.sagemaker_session,
+        if instance_type is None or isinstance(instance_type, str):
+            instance_type_for_volume_kms = instance_type
+        elif isinstance(instance_type, ParameterString):
+            instance_type_for_volume_kms = instance_type.default_value
+        else:
+            raise ValueError(f"Bad value for instance type: '{instance_type}'")
```

Comment:
I'm pretty sure. The unit tests make it seem like it's compatible. There shouldn't be a different python type for the instance type.

Reply:
Today if a user passes in a different python type, they wont get an error right? (My understanding is that the "Optional[Union[str, PipelineVariable]]" is not strictly enforced by Python). Let me know if im mistaken

If Im not wrong about the above, this would be changing behavior of the Estimator class (and I havent seen type assertions like these in other places anyway). Seems safer to not change this behavior

Reply:
Ok, I can skip throwing an exception. However, if a variable that isn't a string or `PipelineVariable` is used, there will probably still be an exception thrown somewhere

Reply:
https://github.com/aws/sagemaker-python-sdk/issues/3917 - let me know how I can help

Reply:
It turns out this is not backward compatible, see the issue https://github.com/aws/sagemaker-python-sdk/issues/3993

Have we made the fix to skip PipelineVariable type?